### PR TITLE
Controller name options for #resource/#resources

### PR DESCRIPTION
**Pull request by gzigzigzeo**
resource 'users', controller: 'profile'

Now produces:

get '/users' to: Web::Controller::Profile::Index
get '/users/new' to: Web::Controller::Profile::New

### CODE REVIEW

**Comment by jodosha** [comment]
@gzigzigzeo Thanks for this PR, can you please elaborate more your use case? Thank you.

**Comment by gzigzigzeo** [comment]
Sure.

namespace 'admin' do
  resources :users, controller: 'admin/users'
end

**Comment by gzigzigzeo** [comment]
@jodosha It is useful with namespacing support merged to utils.

**Comment by jodosha** [comment]
@gzigzigzeo Sorry if I insist, but I'm trying to have a better understanding. By looking at the name of the action it looks like a Lotus convention, so I guess a greenfield project. Those kind of configurations are often required for legacy code.

So why do you have the need of decouple the name of the resource from the name of the controller?

**Comment by gzigzigzeo** [comment]
@jodosha 

- Use case for the lastest example: 

```ruby
resource 'me', only: %i(show edit update), controller: 'profile'
```

- It should not break the integration because I can do the same thing manually, without use #resources:

```ruby
namespace 'admin' do
  resources 'people', controller: 'admin/users'
end
```

is equal to

```ruby
namespace 'admin' do
  get '/people', to: 'admin/users#index'
  get '/people/new', to: 'admin/users#new'
  ...
```

With my merged PR to `lotus/utils` (`#classify` fixes) it works fine.

Sure, convention is important and good. This option does not break the convention, but gives me more control on generated resource routes.

**Comment by AlfonsoUceda** [comment]
if this doesn't break the convention and give us more flexibility then :+1:

**Comment by joneslee85** [comment]
:+1: much thanks

File: lib/lotus/routing/resource/action.rb [hunk 1]
```diff
@@ -221,6 +221,22 @@ def endpoint
...
+        # @example
+        #   Lotus::Router.new do
+        #     resources 'flowers', controller: 'rocks'
+        #   end
+        #
+        #   # It will mount path 'flowers/new' to Rocks::New instead of Flowers::New
+        #   # Same for other action names
+        #
+        # @api private
+        # @since 0.2.1
```
Comment (jodosha) [inline lib/lotus/routing/resource/action.rb]: @gzigzigzeo @joneslee85 this should be `@since x.x.x`

**Comment by jodosha** [comment]
@gzigzigzeo Thank you very much for this. :+1:

**Comment by jodosha** [comment]
@gzigzigzeo @joneslee85 This is a nice addition, but it works only for `resource`. We should add the same functionality for `resources` as well.

**Comment by gzigzigzeo** [comment]
@jodosha

* Can not reproduce it with your snippet:

```ruby
Bundler could not find compatible versions for gem "lotus-utils":
  In Gemfile:
    lotus-controller (~> 0.3) ruby depends on
      lotus-utils (~> 0.4) ruby

    lotusrb (>= 0) ruby depends on
      lotus-utils (0.3.5)
```
I think lotus/lotus dependency on lotus-utils should be changed as follows:

```ruby
  spec.add_dependency 'lotus-utils',      '~> 0.4', '>= 0.4.0'
```

* You have the mistake in your example:

```ruby
resources :bo, controller: 'books', only: :show
```
creates single route path '/bo/:id' (it will work, I have checked it in my current app). To create '/bo' you should specify change the `:only` option as follows:

```ruby
resources :bo, controller: 'books', only: %i(show index)
```

File: lib/lotus/routing/resource/action.rb (hunk 1 above)
Comment (gzigzigzeo) [inline lib/lotus/routing/resource/action.rb]: @jodosha Sorry, I did not understood. I see @since there, am I wrote wrong version number?

**Pull request merged**